### ISSUE TITLE
The XHTML index.html and content files should use dir="rtl" for Arabic content

### ISSUE THREAD

**Issue by raducoravu** [dita-ot/dita-ot#1710]
The DITA Map has the xml:lang="ar" set to it. Each topic also has the language set to Arabic.
The XHTML generated for each topic contains dir="rtl" set on the HTML root.
But for the TOC the XSL:

            DITA-OT/plugins/org.dita.xhtml/xsl/map2htmtoc/map2htmtocImpl.xsl

in the template:
 
         <xsl:template name="generate-toc">

does not generate any attribute for the HTML root element.
The "DITA-OT/plugins/org.dita.xhtml/xsl/xslhtml/dita2htmlImpl.xsl" stylesheet defines an XSL template called "bidi-area" which should probably be moved to a common XSLT and used for generting the TOC list of references as well.

**Comment by robander** [comment]
Tested an Arabic map with XHTML and HTML5 output using 2.2.4. In each case, the lang attribute is set in the TOC output. 

HTML5 TOC output:
```
<!DOCTYPE html
  SYSTEM "about:legacy-compat">
<html lang="ar" dir="rtl"><head>...
```

XHTML TOC output:
```
<?xml version="1.0" encoding="UTF-8"?>
<!DOCTYPE html
  PUBLIC "-//W3C//DTD XHTML 1.0 Transitional//EN" "http://www.w3.org/TR/xhtml1/DTD/xhtml1-transitional.dtd">
<html xmlns="http://www.w3.org/1999/xhtml" xml:lang="ar" lang="ar" dir="rtl">
<head><meta http-equiv="Content-Type" content="text/html; charset=UTF-8" />
...
```

I think this has been resolved.

**Comment by raducoravu** [comment]
So the initial issues says that if xml:lang="ar-ar" is set on the DITA Map root element (and the @dir attribute is not set at all in the DITA content) and you generate HTML output, the @dir="rtl" attribute should also be automatically set on the generated "index.html". I tried with 2.2.4 (HTML5) but I cannot get this to work.

**Comment by robander** [comment]
Ah, I tried "ar" - maybe this is also related to #2267?

**Comment by robander** [comment]
Double checked with 2.2.4 -- `ar` and `ar-eg` both result in RTL output in `index.html`, but `ar-ar` does not.

**Comment by xephon2** [comment]
> **Off Topic**
> The `xml:lang` regional subtag values are token from ISO 3166-1 and should therefore be uppercase. 

:smile: :+1:  `en-US`
:disappointed: :-1:  `en-us`

The region subtag in RFC 3066 took its values from the ISO 3166 country codes, see [w3.org - The region subtag](https://www.w3.org/International/articles/language-tags/#region).

According to ISO 3166-1 the alpha-2 codes always have to be uppercase, see [ISO Online Browsing Platform](https://www.iso.org/obp/ui/#search/code/). This is correct in the DITA-OT docs: [Supported languages: HTML-based transformations](http://www.dita-ot.org/dev/user-guide/DITA-globalization-xhtml.html)

Sometimes language service providers or language processing tools have problems with lowercase subtags.

**Comment by robander** [comment]
In my memory - the aa-BB format is a recommendation but lower case values are not invalid. I see a wide mix of cases and we should continue to support both in source files. 

The string files we ship used lower case values initially because a) it's internal to the tool (not something any user should care about), and 2) it simplified comparison in XSLT (lower-case whatever value is found in the map or topic, then check if we support that value).

**Comment by robander** [comment]
Double checked - the DITA specification has a normative reference to RFC 5646 for language identification: https://www.rfc-editor.org/rfc/rfc5646.txt

From section 2.1.1 in that RFC:
> At all times, language tags and their subtags, including private use
>   and extensions, are to be treated as case insensitive: there exist
>   conventions for the capitalization of some of the subtags, but these
>   MUST NOT be taken to carry meaning.

>   Thus, the tag "mn-Cyrl-MN" is not distinct from "MN-cYRL-mn" or "mN-
>   cYrL-Mn" (or any other combination), and each of these variations 
>   conveys the same meaning: Mongolian written in the Cyrillic script as
>   used in Mongolia.

**Comment by xephon2** [comment]
@robander, thanks a lot for clarification!

**Comment by robander** [comment]
No problem - you're not the first to wonder about that - there's clearly uncertainty around case values in `@xml:lang`.

**Comment by infotexture** [comment]
@robander recalls correctly: the alphabetic region subtags were changed to upper case in dita-ot/docs@217e2bf per the convention in ISO 3166-1.

**Comment by raducoravu** [comment]
Actually I'm using:

    xml:lang="ar-AR"

and that also does not work.
But I do confirm this works:

     xml:lang="ar"

**Comment by robander** [comment]
Interesting - tried on both the map and content with `xml:lang="ar-AR"`. 

- The XHTML `index.html` output file does not set `@dir` (the original issue reported here).
- The XHTML content file also does not set `@dir` on the topic (think this goes to the same part of the code)
- But, the generated text in that topic is correct - the table caption comes out with "جدول 1." rather than "Table 1."

I'm going to update the title here to indicate that it's not just a problem with the TOC.

**Comment by robander** [comment]
Tried this one again with some of the latest 2.3 code. When I use `xml:lang="ar-AR"` in a map and in the referenced topic, the HTML5 and XHTML output both get `dir="rtl"`. That fits with my understanding of how the updated `bidi-area` template should work in `dita-utilities.xsl`. I think this one is actually in 2.3 fixed with the update for #2267.

**Comment by raducoravu** [comment]
Thanks Robert.
